code/modules/spells/spell_types/touch/_touch.dm

```diff
@@ -170,24 +170,59 @@
 	SHOULD_NOT_OVERRIDE(TRUE) // DEFINITELY don't put effects here, put them in cast_on_hand_hit
 
 	if(!can_hit_with_hand(target, caster))
-		return
+		return FALSE
+
+	SEND_SIGNAL(src, COMSIG_SPELL_TOUCH_HAND_HIT, target, caster, source)
+
+	var/mob/mob_victim = target
+	if(istype(mob_victim) && mob_victim.can_block_magic(antimagic_flags))
+		on_antimagic_triggered(source, target, caster)
+
+	else if(!cast_on_hand_hit(source, target, caster))
+		return FALSE
 
-	INVOKE_ASYNC(src, PROC_REF(do_hand_hit), source, target, caster)
```

Comment:
Why did we move all of this code into this function?

Reply:
Because with the invoke async it was making it not possible to read the proper return values.
So it was added to the main proc in order to properly return when the hand hit is completed.